src/power_grid_model_io/utils/uuid_excel_cvtr.py

```diff
@@ -161,9 +169,9 @@ def insert_or_update_number_column(
     new_column_name = guid_column.replace("GUID", number)
     if guid_column == "GUID":
         if sheet_name in special_nodes_en:
-            new_column_name = guid_column.replace("GUID", "Subnumber")
+            new_column_name = guid_column.replace("GUID", VISION_EXCEL_LAN_DICT[LANGUAGE_EN][DICT_KEY_SUBNUMBER])
         elif sheet_name in special_nodes_nl:
-            new_column_name = guid_column.replace("GUID", "Subnummer")
+            new_column_name = guid_column.replace("GUID", VISION_EXCEL_LAN_DICT[LANGUAGE_NL][DICT_KEY_SUBNUMBER])
```

Comment:
cfr before: can we remove this duplication taken into account my previous comments?

Reply:
The problem is that the `guid_column.replace(` only happens under the two conditions. I tried to initialize the `special_key_language` to `''` which corresponds to `.replace('GUID','GUID')`. It fails in some cases unfortunately and I figure to add the extra logic to prevent the `replace` from taking place, it would already be longer code than what is done currentlt.

Reply:
but the other one also only loops over specifically `GUID` columns. if the logic becomes a separate function, the `== GUID` check may be either:

* pushed either outside the function (if the function itself does not have a `for` loop; or
* provided as an input for the function e.g. as `guid_columns=["GUID"]` (if the new function does contain a `for` loop over all `guid_columns`)

Reply:
Right. Solution comitted.